gateway-service/src/main/java/org/zowe/apiml/gateway/filters/pre/HeaderSanitizerFilter.java

```diff
@@ -48,10 +50,35 @@ public boolean shouldFilter() {
     @Override
     public Object run() {
         RequestContext context = RequestContext.getCurrentContext();
-        for (String header : headersToBeCleared) {
-            context.addZuulRequestHeader(header, null);
-        }
+
+        Map<String, String> requestHeaders = new HashMap<>();
+
+        includeHeadersFromZuul(context, requestHeaders);
+        includeHeadersFromRequest(context, requestHeaders);
+
+        nullHeadersOfInterestIntoZuulRequestHeaders(context, requestHeaders);
+
         return null;
     }
 
+    private void nullHeadersOfInterestIntoZuulRequestHeaders(RequestContext context, Map<String, String> requestHeaders) {
+        Arrays.stream(headersToBeCleared).forEach( toBeCleared -> //for each header to be cleared
+            requestHeaders.entrySet().stream() //in all requestHeaders
```

Comment:
can we get header by key here? this solution causes big time complexity

Reply:
We cannot get header by key with case insensitivity. That is what happens in reality in one of the IT.